common/app/Panes/redux/index.js

```diff
@@ -172,19 +166,24 @@ export default function createPanesAspects({ createPanesMap }) {
         const panesMap = action.meta.panesMap;
         const panes = _.map(panesMap, (name, type) => ({ name, type }));
         const numOfPanes = Object.keys(panes).length;
+        const panesByName = _.isEqual(state.panes, panes) && state.panesByName;
         return {
           ...state,
           panesMap,
           panes,
-          panesByName: panes.reduce((panes, { name }, index) => {
-            const dividerLeft = utils.getDividerLeft(numOfPanes, index);
-            panes[name] = {
-              name,
-              dividerLeft,
-              isHidden: name === 'Map' ? state.isMapPaneHidden : false
-            };
-            return panes;
-          }, {})
+          panesByName: panesByName
+            ? panesByName
+            : panes.reduce((panes, { name }, index) => {
+                const dividerLeft = utils.getDividerLeft(numOfPanes, index);
+                panes[name] = {
+                  name,
+                  dividerLeft,
+                  isHidden: false
+                };
+                return panes;
+              },
+            {}
+          )
```

Comment:
@BerkeleyTrue Ok, I think this does it. We can just compare prev panes to new panes, and base the rest on that: if the number or type of panes that exist between transitions doesn't change, don't change or recalculate anything &mdash; just persist `state.panesByName` which preserves all `dividerLeft` values as well as `isHidden`. 

Otherwise, just recalculate `dividerLeft` and set `isHidden` to false for every pane as normal. 

The latter should only happen when new types of panes are introduced (e.g. from Map/Step to challenge), but doesn't couple the state with any type of pane or content in particular.